Assets/MixedRealityToolkit/_Core/Definitions/Utilities/InscribedRectangle.cs

```diff
@@ -0,0 +1,570 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Utilities;
+using System;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Definitions.Utilities
+{
+    /// <summary>
+    /// The InscribedRectangle class defines the largest rectangle within an
+    /// arbitrary shape.
+    /// </summary>
+    public class InscribedRectangle
+    {
+        /// <summary>
+        /// Total number of starting points randomly generated within the boundary.
+        /// </summary>
+        private static readonly int randomPointCount = 30;
+
+        /// <summary>
+        /// The total amount of height, in meters,  we want to gain with each binary search
+        /// change before we decide that it's good enough.
+        /// </summary>
+        private const float minimumHeightGain = 0.01f;
```

Comment:
What's our naming conventions for private const values?

Reply:
i don't think we have established one. i took this from the june18_dev PR i ported